ReactiveCocoa/Swift/SignalProducer+Restart.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SignalProducer+Restart.swift
```

Comment:
That's really minor, but if we would really want to merge this into upstream RAC, this operator should be placed in `SignalProducer.swift` where it really belongs to. Same applies for the spec that's currently sits in a test project. It would fit nicely in this PR if added to `SignalProducerSpec.swift`.